deepmd/fit/polar.py

```diff
@@ -303,72 +307,168 @@ def build (self,
+                    if (not self.uniform_seed) and (self.seed is not None): self.seed += self.seed_shift
+                    # (nframes x natoms) x naxis
+                    final_layer = tf.reshape(final_layer, [tf.shape(inputs)[0] * natoms[2+type_i], self.dim_rot_mat_1])
+                    # (nframes x natoms) x naxis x naxis
+                    final_layer = tf.matrix_diag(final_layer)
                 else :
-                    layer = one_layer(layer, self.n_neuron[ii], name='layer_'+str(ii)+'_type_'+str(type_i)+suffix, reuse=reuse, seed = self.seed, activation_fn = self.fitting_activation_fn, precision = self.fitting_precision, uniform_seed = self.uniform_seed, initial_variables = self.fitting_net_variables, mixed_prec = self.mixed_prec)
+                    bavg = np.zeros(self.dim_rot_mat_1*self.dim_rot_mat_1)
+                    # bavg[0*self.dim_rot_mat_1+0] = self.avgeig[0]
+                    # bavg[1*self.dim_rot_mat_1+1] = self.avgeig[1]
+                    # bavg[2*self.dim_rot_mat_1+2] = self.avgeig[2]
+                    # (nframes x natoms) x (naxis x naxis)
+                    final_layer = one_layer(layer, self.dim_rot_mat_1*self.dim_rot_mat_1, activation_fn = None, name='final_layer_type_'+str(type_i)+suffix, reuse=reuse, seed = self.seed, bavg = bavg, precision = self.fitting_precision, uniform_seed = self.uniform_seed, initial_variables = self.fitting_net_variables, mixed_prec = self.mixed_prec, final_layer = True)
+                    if (not self.uniform_seed) and (self.seed is not None): self.seed += self.seed_shift
+                    # (nframes x natoms) x naxis x naxis
+                    final_layer = tf.reshape(final_layer, [tf.shape(inputs)[0] * natoms[2+type_i], self.dim_rot_mat_1, self.dim_rot_mat_1])
+                    # (nframes x natoms) x naxis x naxis
+                    final_layer = final_layer + tf.transpose(final_layer, perm = [0,2,1])
+                # (nframes x natoms) x naxis x 3(coord)
+                final_layer = tf.matmul(final_layer, rot_mat_i)
+                # (nframes x natoms) x 3(coord) x 3(coord)
+                final_layer = tf.matmul(rot_mat_i, final_layer, transpose_a = True)
+                # nframes x natoms x 3 x 3
+                final_layer = tf.reshape(final_layer, [tf.shape(inputs)[0], natoms[2+type_i], 3, 3])
+                # shift and scale
+                sel_type_idx = self.sel_type.index(type_i)
+                final_layer = final_layer * self.scale[sel_type_idx]
+                final_layer = final_layer + self.constant_matrix[sel_type_idx] * tf.eye(3, batch_shape=[tf.shape(inputs)[0], natoms[2+type_i]], dtype = GLOBAL_TF_FLOAT_PRECISION)
```

Comment:
I wonder `dtype` is `self.fitting_precision` instead of `GLOBAL_TF_FLOAT_PRECISION`.